### PR TITLE
namcoic.cpp : Simpler bit-swapped values

### PR DESCRIPTION
**Pull request by cam900**
namcoic.cpp : Reduce duplicate

### CODE REVIEW

File: src/mame/machine/namcoic.cpp [hunk 1]
```diff
@@ -867,62 +867,42 @@ READ16_MEMBER( namcos2_shared_state::c355_obj_ram_r )
...
 			break;
 
 		case NAMCONB2_OUTFOXIES:
 			bank = nth_byte16(&m_c169_roz_bank[which * 8 / 2], (tile >> 11) & 0x7);
-			tile = (tile & 0x7ff) | (bank * 0x800);
-			mangle = tile & ~0x50;
-			if (tile & 0x10) mangle |= 0x40;
-			if (tile & 0x40) mangle |= 0x10;
+			tile = (tile & 0x7ff) | (bank << 11);
+			mangle = (bitswap<7>(tile & 0x50, 4, 5, 6, 3, 2, 1, 0)) | (tile & ~0x50);
```
Comment (galibert) [inline src/mame/machine/namcoic.cpp]: You should turn that into one unique bitswap without masking

**Comment by DavidHaywood** [comment]
are some of these things that could instead be solved by more logical ROM loading or lineswaps on startup? (ie maybe the roms were loaded in a way to try and save memory before, rather than mirroring or leaving gaps)

(disclaimer, I haven't looked but per-game cases in drawing code are unexpected if the video hardware is using same custom chips)

**Comment by cam900** [comment]
I think these are address bit-swapped, and It's for make bitswap functions simpler.

**Comment by DavidHaywood** [comment]
Ok, well, it might make more sense to start handling this kind of thing with callbacks added in the machine config like tile banking on data east games etc. if you're looking for more cleanup opportunity

having large per-game enum statements and per-game switch cases, especially in common device implementations is generally considered to be bad form now that MAME is a proper c++ project and since the Namco drivers haven't really had an active 'owner' for a long time there are a lot of them still doing it.

File: src/mame/drivers/namcos2.cpp [hunk 2]
```diff
@@ -5904,8 +5896,8 @@ GAME(  1988, assault,    0,        base2,    assault,  namcos2_state, init_assau
 GAME(  1988, assaultj,   assault,  base2,    assault,  namcos2_state, init_assaultj,      ROT90, "Namco", "Assault (Japan)", 0 )
 GAME(  1988, assaultp,   assault,  assaultp, assault,  namcos2_state, init_assaultp_hack, ROT90, "Namco", "Assault Plus (Japan)", 0)
 
-GAME(  1988, metlhawk,   0,        metlhawk, metlhawk, namcos2_state, init_metlhawk, ROT90,  "Namco", "Metal Hawk (Rev C)", 0)
-GAME(  1988, metlhawkj,  metlhawk, metlhawk, metlhawk, namcos2_state, init_metlhawk, ROT90,  "Namco", "Metal Hawk (Japan, Rev F)", 0)
+GAME(  1988, metlhawk,   0,        metlhawk, metlhawk, namcos2_state, init_metlhawk, ROT90,  "Namco", "Metal Hawk (Rev C)", MACHINE_IMPERFECT_GRAPHICS )
+GAME(  1988, metlhawkj,  metlhawk, metlhawk, metlhawk, namcos2_state, init_metlhawk, ROT90,  "Namco", "Metal Hawk (Japan, Rev F)", MACHINE_IMPERFECT_GRAPHICS )
```
Comment (cuavas) [inline src/mame/drivers/namcos2.cpp]: I'm not really sure it's worth adding an imperfect graphics flag for this.  The issues are relatively minor, and they're noted here in the source.

**Comment by Tafoid** [comment]
Something is amiss here in my snapshot compares (10 seconds run):

finalap2, finalap3, finalap3a - namcos2.cpp
finalap2-diff

finalap3-diff
finalap3a-diff

**Comment by Osso13** [comment]
Also breaks finallap title screen

**Pull request merged**